tools/serverpod_cli/lib/src/generator/dart/library_generators/model_library_generator.dart

```diff
@@ -152,8 +152,7 @@ class SerializableModelLibraryGenerator {
           parentClass.className,
           p.joinAll([
             'package:${serverCode ? '${config.serverPackage}/src/generated' : '${config.dartClientPackage}/src/protocol'}',
-            ...parentClass.subDirParts,
-            '${parentClass.fileName}.dart',
+            'protocol.dart',
```

Comment:
Suggestion: We already have code that manages these path inside the class `TypeDefinition`. But currently we only store this type definition for the class fields, but not for the class it self. 

This will be a little annoying to do, but I think it is the correct way to handle this case. 

Introduce a new field `type` on the `SerializableModelDefinition` class. In the model parser `e.g. model_parser.dart` you have to parse the type with the method `parseType` as the input of the type you should use `$defaultModulAlias:$className`.

e.g.
```dart
var classType = parseType(
      '$defaultModuleAlias:$className',
      extraClasses: extraClasses,
    );
```

Then you need to pass this type on into your parent resolver class you created. 

Then you should be able to call the method `reference` on the type to get the full URL automagically, instead of constructing it manually here.

E.g.

```dart
parentClass.type.reference(
          serverCode,
          subDirParts: subDirParts,
          config: config,
        );
```

Reply:
With this, it would be nice with a small little test for the type field on the `SerializableModelDefinition` when parsing a yaml file. To verify that we create it correctly, will be enough to check that the class name is the type name.